includes/class-sensei-preview-user.php

```diff
@@ -258,6 +288,97 @@ private function create_role() {
 		}
 	}
 
+	/**
+	 * Allow preview user to view draft posts.
+	 *
+	 * This effectively allows them the 'edit_post' and 'read_private_posts' caps, but this filter will only run on course frontend pages.
+	 *
+	 * @note This hook should only run when the preview user is active, it does not do checks on its own.
+	 *
+	 * @access private
+	 *
+	 * @param array  $caps    Capabilities.
+	 * @param string $cap     Capability.
+	 * @param int    $user_id User ID.
+	 * @param array  $args    Arguments.
+	 *
+	 * @return array
+	 */
+	public function allow_post_preview( $caps, $cap, $user_id, $args ) {
+
+		if ( get_current_user_id() !== $user_id ) {
+			return $caps;
+		}
+
+		if ( in_array( $cap, [ 'edit_post', 'read_private_posts' ], true ) ) {
+			return [];
```

Comment:
Loved this 👍

Reply:
Mostly everything worked as expected. One small issue that I've found is if I have multiple draft lessons in a draft course, clicking complete on any of the lessons takes me to the Course Completion page

Reply:
Trying to reproduce this but it's working for me. Could you check what numbers the course progress block says after completing just one lesson? 

<img width="711" alt="image" src="https://user-images.githubusercontent.com/176949/211353144-fc8134b9-b587-477c-8240-fecb9ba33faf.png">

Interestingly it does take me to the course completed page after I click Exit Student Preview.

Reply:
Can't reproduce it again either, not sure what happened back then. Not seeing the completion page now. So all good 👍 If I find any exact scenario or an edge case where it happens, will let you know ☺️